ServerCore/ServerMessageListener.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.AspNetCore.SignalR.Client;
+using Microsoft.Extensions.Hosting;
+
+namespace ServerCore
+{
+    /// <summary>
+    /// Singleton that registers listeners for all server messages to distribute them to listening components
+    /// </summary>
+    public class ServerMessageListener : IDisposable
+    {
+        HubConnection HubConnection { get; set; }
+        IDisposable exampleSubscription = null;
+
+        public ServerMessageListener(IHubContext<ServerMessageHub> hub, IWebHostEnvironment env)
+        {
+            string localhostSignalRUrl;            
+            if (env.IsDevelopment())
+            {
+                localhostSignalRUrl = "http://localhost:44319/serverMessage";
+            }
+            else
+            {
+                localhostSignalRUrl = "http://localhost/serverMessage";
+            }
+
+            HubConnection = new HubConnectionBuilder().WithUrl(localhostSignalRUrl).WithAutomaticReconnect().Build();
+
+            // Register listeners
+            var onExampleMessage = OnExampleMessage;
+            exampleSubscription = HubConnection.On(nameof(ExampleMessage), onExampleMessage);
+
+            TryInitAsync(hub).Wait();
```

Comment:
This is because connecting from the Listener to the Hub doesn't work until the Hub is fully initialized as part of ASP.NET Core startup. I haven't been able to find a way to trigger this on Hub initialization. Instead, this takes advantage of not getting initialized until the first page view. I looked at a way to integrate into startup, but couldn't figure it out. Are you ok with starting with this and seeing if we have problems?

Reply:
I was just wondering if you could store the result of TryInitAsync in a Task, and then await that task before broadcasting a message in ServerMessageHub, since the broadcaster is properly async.

Reply:
Discussed offline and moving to an EnsureInitializedAsync pattern in a new PR